src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java

```diff
@@ -56,43 +67,30 @@ public String getName() {
 
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
-        SqlRequest sqlRequest;
-        try {
-            sqlRequest = SqlRequestFactory.getSqlRequest(request);
-        } catch(IllegalArgumentException e) {
-            // FIXME: need to send proper error response to client.
-            LOG.error("Failed to parse SQL request.", e);
-            return null;
-        }
-
         try {
-            SearchDao searchDao = new SearchDao(client);
-            QueryAction queryAction= null;
-
-            queryAction = searchDao.explain(sqlRequest.getSql());
+            final SqlRequest sqlRequest = SqlRequestFactory.getSqlRequest(request);
+            final QueryAction queryAction = new SearchDao(client).explain(sqlRequest.getSql());
             queryAction.setSqlRequest(sqlRequest);
 
-            // TODO add unittests to explain. (rest level?)
             if (request.path().endsWith("/_explain")) {
                 final String jsonExplanation = queryAction.explain().explain();
-                return channel -> channel.sendResponse(new BytesRestResponse(RestStatus.OK, jsonExplanation));
+                return channel -> channel.sendResponse(new BytesRestResponse(OK, jsonExplanation));
             } else {
                 Map<String, String> params = request.params();
                 RestExecutor restExecutor = ActionRequestRestExecutorFactory.createExecutor(params.get("format"), queryAction);
-                final QueryAction finalQueryAction = queryAction;
                 //doing this hack because elasticsearch throws exception for un-consumed props
                 Map<String,String> additionalParams = new HashMap<>();
                 for (String paramName : responseParams()) {
                     if (request.hasParam(paramName)) {
                         additionalParams.put(paramName, request.param(paramName));
                     }
                 }
-                return channel -> restExecutor.execute(client, additionalParams, finalQueryAction, channel);
+                return channel -> restExecutor.execute(client, additionalParams, queryAction, channel);
             }
-        } catch (SqlParseException | SQLFeatureNotSupportedException e) {
+        } catch (Exception e) {
             // FIXME: need to catch all exceptions to avoid ES process from crashing
```

Comment:
This ```FIXME``` can be removed if it's addressed by the change to ```catch (Exception e)``` above.

Reply:
Yes, removing it. Thanks!